CMakeLists.txt

```diff
@@ -168,32 +170,22 @@ endif (HAS_ROCM EQUAL 0)
 # Detect AMDGPU backend for native codegen
 #################
 
-set(HSA_AMDGPU_GPU_TARGET "AMD:AMDGPU:8:0:3" CACHE STRING "Target AMDGPU ISA version")
+set(HSA_AMDGPU_GPU_TARGET "gfx803" CACHE STRING "Target AMDGPU ISA version")
 
 # Tokenize HSA_AMDGPU_GPU_TARGET
 foreach(AMDGPU_GPU_TARGET IN LISTS HSA_AMDGPU_GPU_TARGET)
-  if (NOT (AMDGPU_GPU_TARGET STREQUAL "carrizo" OR
-           AMDGPU_GPU_TARGET STREQUAL "fiji" OR
-           AMDGPU_GPU_TARGET STREQUAL "hawaii" OR
-           AMDGPU_GPU_TARGET STREQUAL "AMD:AMDGPU:7:0:1" OR
-           AMDGPU_GPU_TARGET STREQUAL "AMD:AMDGPU:8:0:1" OR
-           AMDGPU_GPU_TARGET STREQUAL "AMD:AMDGPU:8:0:3"))
-    MESSAGE(FATAL_ERROR "${AMDGPU_GPU_TARGET} is not a valid AMDGPU target")
-  endif()
-
-  # AMDGPU_TARGET_LIST would hold values used by compiler scripts
-  # Map ISA version to GPU family
-  if (AMDGPU_GPU_TARGET STREQUAL "AMD:AMDGPU:7:0:1")
+  # AMDGPU_TARGET_LIST holds the values which are used by the compiler scripts.
+  if (AMDGPU_GPU_TARGET STREQUAL "gfx701")
     list(APPEND AMDGPU_TARGET_LIST "hawaii")
-  endif()
-
-  if (AMDGPU_GPU_TARGET STREQUAL "AMD:AMDGPU:8:0:1")
+  elseif (AMDGPU_GPU_TARGET STREQUAL "gfx801")
     list(APPEND AMDGPU_TARGET_LIST "carrizo")
-  endif()
-
-  if (AMDGPU_GPU_TARGET STREQUAL "AMD:AMDGPU:8:0:3")
+  elseif (AMDGPU_GPU_TARGET STREQUAL "gfx802")
+    list(APPEND AMDGPU_TARGET_LIST "tonga")
+  elseif (AMDGPU_GPU_TARGET STREQUAL "gfx803")
     list(APPEND AMDGPU_TARGET_LIST "fiji")
```

Comment:
Excellent idea, I wish I had thought about that before! Please see the updated form and if it fits what you had in mind. It is not necessarily the most efficient way of doing things, but I presume we will remove this sooner as opposed to later. Thank you for the suggestion.